reco_utils/recommender/tfidf/tfidf_utils.py

```diff
@@ -0,0 +1,400 @@
+            verbose (boolean): True or False for whether to print.
+        
+        Returns:
+            clean (str): Cleaned version of text.
+        """
+        
+        try:
+            # Normalize unicode
+            text_norm = unicodedata.normalize('NFC', text)
+
+            # Remove HTML tags
+            clean = re.sub('<.*?>', '', text_norm)
+
+            # Remove new line and tabs
+            clean = clean.replace('\n', ' ')
+            clean = clean.replace('\t', ' ')
+            clean = clean.replace('\r', ' ')
+            clean = clean.replace('Â\xa0', '')     # non-breaking space
+
+            # Remove all punctuation and special characters
+            clean = re.sub('([^\s\w]|_)+','', clean)
+
+            # If you want to keep some punctuation, see below commented out example
+            # clean = re.sub('([^\s\w\-\_\(\)]|_)+','', clean)
+
+            # Skip further processing if the text will be used in BERT tokenization
+            if for_BERT is False:
+                # Lower case
+                clean = clean.lower()
+        except:
+            if verbose is True:
+                print('Cannot clean non-existent text')
+            clean = ''
+        
+        return clean
+
+    def clean_dataframe(self, df, cols_to_clean, new_col_name='cleaned_text'):
+        """ Clean the text within the columns of interest and return a dataframe with cleaned and combined text.
+        
+        Args:
+            df (pd.DataFrame): Dataframe containing the text content to clean.
+            cols_to_clean (str): List of columns to clean by name (e.g., ['abstract','full_text']).
+            new_col_name (str): Name of the new column that will contain the cleaned text.
+
+        Returns:
+            df (pd.DataFrame): Dataframe with cleaned text in the new column.
+        """
+        # Check if for BERT tokenization
+        if self.tokenization_method in ['bert','scibert']:
+            for_BERT = True
+        else:
+            for_BERT = False
+
+        # Clean the text in the dataframe
+        for i in range(0, len(df)):
+            for col in cols_to_clean:
+                df[col][i] = self.__clean_text(df[col][i], for_BERT)
+
+        # Collapse the table such that all descriptive text is just in a single column
+        first_col = True
+        for col in cols_to_clean:
+            if first_col is True:
+                # first one
+                df[new_col_name] = df[col]
+                first_col = False
+            else:
+                df[new_col_name] = df[new_col_name] + ' ' + df[col]
+        
+        # Make sure any punctuation or special characters in Name are human readible
+        for i in range(0, len(df)):
+            df[self.title_col][i] = df[self.title_col][i].encode('ascii','ignore')
+        
+        for i in range(0, len(df)):
+            df[self.title_col][i] = df[self.title_col][i].decode()
```

Comment:
```suggestion
        df[self.title_col] = df[self.title_col].apply(lambda x: x.encode("ascii", "ignore").decode())
```

Reply:
These changes have been added in the latest commit.